api/app.py

```diff
@@ -194,3 +194,51 @@ def get_assignment_id(category_type: str, assignment_number: int, lab_type: int
 
     # Return assignment ID if found for categories other than "labs"
     return {"assignment_id": assignment_data.get("assignment_id", "Assignment ID not found.")}
+
+
+@app.get("/fetchAllGrades")
+@handle_errors
+def fetchAllGrades(class_id: str = None):
+    """
+    Fetch Grades for all assignments for all students
+
+    Parameters:
+    - class_id (str, optional): The ID of the class for which assignments are being retrieved. 
+      Defaults to `None`.
+
+    Returns:
+    - JSON
+
+    Example Output:
+    {
+        "5211610": [
+            {
+            "Name": "test2",
+            "SID": "",
+            "Email": "test2@test.com",
+            "Total Score": "",
+            "Max Points": "4.0",
+            "Status": "Missing",
+            "Submission ID": null,
+            "Submission Time": null,
+            "Lateness (H:M:S)": null,
```

Comment:
Lateness is useful for status tracking.
If a student takes an incomplete, what is the average amount of time they submit a late assignment for?
How many assignments are late?